asr-python-advanced-finetune-am-conformer-ctc-tao-finetuning-librispeech-nigerian.ipynb

```diff
@@ -0,0 +1,971 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Install modules that the downloading and preprocessing script requires which aren't part of the Python Standard Library."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install tqdm\n",
+    "!pip install sox"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python ./get_librispeech_data.py --data_root=$HOST_DATA_DIR --data_sets='train_clean_100,dev_clean'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The filepaths in the manifest files are currently set with respect to the system on which we're running this notebook. They need to be set with respect to the `tao` container. The `sed` command will help us fix that. \n",
+    "\n",
+    "Note the use of double quotes in calling `sed`. Those are necessary for `sed` to evaluate `$HOST_DATA_DIR` and `$DATA_DIR`. If you attempt to use single quotes, `sed` will replace the literal string `$HOST_DATA_DIR` with the literal string `$DATA_DIR` instead of replacing the value assigned to `HOST_DATA_DIR` with the value assigned to `DATA_DIR`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!sed -i \"s|$HOST_DATA_DIR|$DATA_DIR|g\" $HOST_DATA_DIR/LibriSpeech/train_clean_100_manifest.json\n",
+    "!sed -i \"s|$HOST_DATA_DIR|$DATA_DIR|g\" $HOST_DATA_DIR/LibriSpeech/dev_clean_manifest.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Remove the `.tar.gz` archive files to save space."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!rm $HOST_DATA_DIR/train_clean_100.tar.gz\n",
+    "!rm $HOST_DATA_DIR/dev_clean.tar.gz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's listen to a sample audio file from the preprocessed training dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# change path of the file here\n",
+    "import os\n",
+    "import IPython.display as ipd\n",
+    "path = os.environ[\"HOST_DATA_DIR\"] + '/LibriSpeech/train-clean-100-processed/163-121908-0000.wav'\n",
+    "ipd.Audio(path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Crowdsourced high-quality Nigerian English speech dataset\n",
+    "The evalutation/finetuning data is publicly available in several files [here](https://www.openslr.org/resources/70/) and can be downloaded directly."
```

Comment:
```suggestion
    "The evalutation/fine-tuning data is publicly available in several files [here](https://www.openslr.org/resources/70/) and can be downloaded directly."
```

Reply:
Just a nit but this isn't directly downloaded like the previous one. The previous one, you click on the link and you specify where you want the .tar.gz file to be saved. Here, you are just taken to a page that provides a list of individual downloads.

Reply:
OK, I've deleted "and can be downloaded directly."